src/worker/bpf.rs

```diff
@@ -0,0 +1,140 @@
+use std::{
+    cmp,
+    ffi::{c_char, CString},
+    fmt::Display,
+    mem, slice, thread, time,
+};
+
+use core_affinity::CoreId;
+use libc::{SYS_bpf, SYS_perf_event_open};
+use log::info;
+
+use aya_obj::copy_instructions;
+use aya_obj::generated::{
+    bpf_attach_type, bpf_attr, bpf_cmd, bpf_prog_type, perf_event_attr,
+    perf_event_sample_format, perf_type_id,
+};
+
+use crate::{BaseConfig, Worker, WorkerError, Workload, WorkloadConfig};
+
+#[derive(Debug, Clone, Copy)]
+pub struct BpfWorker {
+    config: BaseConfig,
+    workload: WorkloadConfig,
+}
+
+impl BpfWorker {
+    pub fn new(workload: WorkloadConfig, cpu: CoreId, process: usize) -> Self {
+        BpfWorker {
+            config: BaseConfig { cpu, process },
+            workload,
+        }
+    }
+}
+
+impl Worker for BpfWorker {
+    fn run_payload(&self) -> Result<(), WorkerError> {
+        info!("{self}");
+
+        let Workload::Bpf { nprogs, tracepoint } = self.workload.workload
+        else {
+            unreachable!()
+        };
+
+        for i in 0..nprogs {
+            // Prepare a bpf program attributes
+            let mut attr = unsafe { mem::zeroed::<bpf_attr>() };
+            let u = unsafe { &mut attr.__bindgen_anon_3 };
+            let prog_fd;
+
+            let mut name: [c_char; 16] = [0; 16];
+            let idx = i.to_string();
+            let cstring = CString::new(format!("berserker{idx}")).unwrap();
+            let name_bytes = cstring.to_bytes();
+            let len = cmp::min(name.len(), name_bytes.len());
+            name[..len].copy_from_slice(unsafe {
+                slice::from_raw_parts(name_bytes.as_ptr() as *const c_char, len)
+            });
+            u.prog_name = name;
```

Comment:
Wonder if this would work:
```rust
// name can be 15 characters at most, so: 15 - len("berserker") = 6
let name = CString::new(format!("berserker{i:.6}")).unwrap();
u.prog_name = name;
```

Reply:
Looks like it won't work, since it produces `CString`, where bpf_attr expects `[i8; 16]` as a name.

Reply:
Right, I see you specifically need a slice with that size now. I'm sure there's a way to simplify this a bit (maybe with iterators?), but I'm not smart enough and it's ok as is.